pubtools/exodus/_tasks/push.py

```diff
@@ -0,0 +1,127 @@
+            help="The exodus-rsync prefix to use (e.g., 'exodus:/mnt/redhat)",
+        )
+
+    def create_publish(self):
+        url = urljoin(self.gw_url, "%s/publish" % self.gw_env)
+        LOG.debug("Creating publish %s", url)
+        response = self.session.post(
+            url,
+            headers={"Content-Type": "application/json"},
+        )
+        response.raise_for_status()
+        task = response.json()
+        return task
+
+    def commit_publish(self):
+        """
+        Commits an exodus-gw publish, e.g.,
+        https://exodus-gw.example.com/prod/publish/4e59c1a0/commit
+        """
+        url = urljoin(
+            self.gw_url,
+            "%s/publish/%s/commit" % (self.gw_env, self.publish.get("id")),
+        )
+        LOG.debug("Committing publish %s", url)
+        response = self.session.post(
+            url,
+            headers={"Content-Type": "application/json"},
+        )
+        response.raise_for_status()
+        task = response.json()
+        return task
+
+    def poll_publish(self, publish_task, tries=4):
+        """Poll the exodus-gw publish task until the task completes, fails, or times out."""
+        wait_time = 0
+        task_url = urljoin(self.gw_url, publish_task["links"]["self"])
+        for i in range(tries):
+            response = self.session.get(task_url, timeout=10)
+            response.raise_for_status()
+            task = response.json()
+            if task["state"] not in ("FAILED", "COMPLETE"):
+                if i < tries - 1:
+                    wait_time = (i + 1) * 30 if wait_time < 120 else wait_time
+                    LOG.info(
+                        "Exodus publish in progress. "
+                        "Will repoll in %d seconds [try %s/%s]",
+                        wait_time,
+                        i + 1,
+                        tries,
+                    )
+                    time.sleep(wait_time)
+                    continue
+                raise RuntimeError("Exodus publish timed out")
+            if task["state"] == "FAILED":
+                raise RuntimeError("Exodus publish failed")
+            if i != 0:
+                LOG.info("Exodus publish succeeded [try: %s/%s]", i + 1, tries)
+            return task
+
+    def run(self):
+        LOG.debug("Exodus push begins")
+
+        if not self.publish:
+            self.publish = self.create_publish()
+        publish_id = self.publish.get("id")
+        LOG.debug("Publish ID: %s", publish_id)
+
+        self.src = self.args.source
+        self.dest = self.args.dest
+        self.prefix = self.args.prefix
+
+        for item in self.src:
+            cmd = [
+                "rsync",
+                "--exodus-publish",
+                publish_id,
+                item,
+                "%s%s%s"
+                % (self.prefix, "" if ":" in self.prefix else ":", self.dest),
+            ]
+            LOG.info(cmd)
+            subprocess.check_output(cmd, shell=False)
```

Comment:
Was it intentional to use check_output and not capture the output from the command?  It means stdout is being discarded.

Actually, it won't work anyway because check_output appeared in python 2.7 but we still need this to work on 2.6 for the time being.